packages/x-data-grid-pro/src/components/GridDetailPanelToggleCell.tsx

```diff
@@ -19,8 +29,17 @@ const useUtilityClasses = (ownerState: OwnerState) => {
   return composeClasses(slots, getDataGridUtilityClass, classes);
 };
 
+const isExpandedSelector = createSelectorMemoizedV8(
+  gridDetailPanelExpandedRowIdsSelector,
+  (expandedRowIds, rowId: GridRowId) => {
+    return expandedRowIds.includes(rowId);
+  },
+);
+
 function GridDetailPanelToggleCell(props: GridRenderCellParams) {
-  const { id, value: isExpanded } = props;
+  const { id, row, api } = props;
+  const rowId = api.getRowId(row);
+  const isExpanded = useGridSelectorV8({ current: api as GridApiPro }, isExpandedSelector, rowId);
```

Comment:
Turned this into a selector with argument. Will conflict with #15200, need to coordinate the merge, cc @MBilalShafi.

Reply:
And it's memoized because `.includes()` is an `O(n)` operation. In the very worst but possible scenario, a user could have all detail panel open, in which case running `.includes()` on each row would be expensive.

Reply:
Looks like a perfect use case for a `Set`, I'll open a PR for v8

Reply:
@lauri865 #15200 has been merged, kindly rebase with master, you could replace `useGridSelectorV8` with `useGridSelector`.

Reply:
done